hw2.py

```diff
@@ -0,0 +1,74 @@
+class Product:
+    def __init__(self, name: str, price: float, unit: float = 1) -> None:
+        self.name = name
+        self.price = price
+        self.unit = unit
+
+    def __repr__(self) -> str:
+        return self.name
+
+    def __float__(self) -> float:
+        return self.price
+
+    def __add__(self, other):
+        if self.name == other.name and self.price == other.price:
+            self.unit += other.unit
+            return self
+        else:
+            cart_new = ShoppingCart()
+            cart_new.add_product(self)
+            cart_new.add_product(other)
+            return cart_new
+
+    def get_total(self, quantity: float = None) -> float:
+        if quantity is None:
+            quantity = self.unit
+        return round(self.price * quantity, 2)
+
+
+class ShoppingCart:
+
+    def __init__(self):
+        self.products: list[Product] = []
+        self.quantities: list[Product] = []
+
+    def __repr__(self) -> str:
+        return "Корзина"
+
+    def __float__(self) -> float:
+        return self.get_total()
+
+    def add_product(self, product, quantity: float = None) -> None:
+        if quantity is None:
+            quantity = product.unit
+        for i in self.products:
```

Comment:
То зробити легше перевіркою
```suggestion
        if product in self.products:
            idx = self.products.index(product)
            self.quantities[idx] += quantity
```

Reply:
Так, може й простіше